### DISCUSSION TITLE
OKD on BareMetal: Failed to push image when creating APP

### DISCUSSION THREAD

**Discussion by okdlukas** [okd-project/okd#1923] [Q&A]
When i want to create any Test Project, for instance the   java-springboot-basic-1-build the created Pod throws this message: 

```
Copying blob sha256:f8837fb50b5a1d8d966d4b33e18f3a31deee3eb7be3a9e0470ba2f80c348a6c4
Copying blob sha256:122eac69054b22f81a29f37eb7effe0a3038861b977db932717c5e068f649107
Copying blob sha256:06f86e50a0b74ff9eb161a7d781228877c90e8ff57e9689e8cb8b0f092a2a9f9
Warning: Push failed, retrying in 5s ...
Registry server Address:
Registry server User Name: serviceaccount
Registry server Email: serviceaccount@example.org
Registry server Password: <<non-empty>>
error: build error: Failed to push image: trying to reuse blob sha256:06f86e50a0b74ff9eb161a7d781228877c90e8ff57e9689e8cb8b0f092a2a9f9 at destination: pinging container registry image-registry.openshift-image-registry.svc:5000: Get "https://image-registry.openshift-image-registry.svc:5000/v2/": dial tcp 172.30.170.198:5000: connect: connection refused

```

I get a PULLBACK Image with this Event on the replicas:
`Failed to pull image "image-registry.openshift-image-registry.svc:5000/okd-test/java-springboot-basic:latest": reading manifest latest in image-registry.openshift-image-registry.svc:5000/okd-test/java-springboot-basic: manifest unknown
`

The Image Registry of the Cluster did this but i don't know if it is related: 

`Failed to update lock: Operation cannot be fulfilled on leases.coordination.k8s.io "openshift-master-controllers": the object has been modified; please apply your changes to the latest version and try again`

When i go into editing deployment i have this marked in red: 
`imagestreamtags.image.openshift.io "java-springboot-basic:latest" not found`

As this are suppose to be the official connected Projects for OKD i would assume that they should be running if created via ADD in Development Mode from OKD. As OKD itself does not give me any clues about the Image-registry connection issue, what can i do to solve it?

**Comment by JaimeMagiera** [comment]
Did you enable the internal registry? Have you verified that it's up and running? 

https://docs.okd.io/4.14/registry/configuring-registry-operator.html

**Reply by okdlukas** [reply]
Did that. The solution was to not use the officially Test projects but instead look into the projects sites and take the links there instead. Now everything is running as it should.
